temporal_sqlalchemy/bases.py

```diff
@@ -109,6 +115,10 @@ def record_history(self,
                 changes = attributes.get_history(clocked, prop.key)
 
             if changes.added:
+                if is_strict_mode:
+                    assert not is_vclock_unchanged, \
+                        'flush() has triggered for a changed temporalized property outside of a clock tick'
+
```

Comment:
I added [test_allow_flushes_within_clock_ticks_when_strict_but_no_change()](https://github.com/CloverHealth/temporal-sqlalchemy/pull/24/files#diff-a11f94dad0b6579a4d552a5a5fafae4bR331) to show why it needs to be in the loop.   I was assuming that it should only throw an exception _if_ a change actually occurred.  

Does this align with your thinking too?

If not, let me know and I can drop the test and move the check outside of the loop.